filament/backend/src/opengl/OpenGLDriver.cpp

```diff
@@ -1789,28 +1822,35 @@ void OpenGLDriver::detachStream(GLTexture* t) noexcept {
     }
 
     GLStream* s = static_cast<GLStream*>(t->hwStream);
-    if (s->isNativeStream()) {
-        mPlatform.detach(t->hwStream->stream);
-        // this deletes the texture id
+    switch (s->streamType) {
+        case StreamType::NATIVE:
+            mPlatform.detach(t->hwStream->stream);
+            break;
+        case StreamType::TEXID:
+            glGenTextures(1, &t->gl.id);
+            break;
+        case StreamType::ACQUIRED:
+            break;
     }
-    glGenTextures(1, &t->gl.id);
+
     t->hwStream = nullptr;
 }
 
 UTILS_NOINLINE
 void OpenGLDriver::replaceStream(GLTexture* t, GLStream* hwStream) noexcept {
     GLStream* s = static_cast<GLStream*>(t->hwStream);
-    if (s->isNativeStream()) {
-        mPlatform.detach(t->hwStream->stream);
-        // this deletes the texture id
-    }
-
-    if (hwStream->isNativeStream()) {
-        glGenTextures(1, &t->gl.id);
-        mPlatform.attach(hwStream->stream, t->gl.id);
-    } else {
-        assert(t->target == SamplerType::SAMPLER_EXTERNAL);
-        t->gl.id = hwStream->user_thread.read[hwStream->user_thread.cur];
+    switch (s->streamType) {
+        case StreamType::NATIVE:
+            mPlatform.detach(t->hwStream->stream);
+            glGenTextures(1, &t->gl.id);
+            mPlatform.attach(hwStream->stream, t->gl.id);
+            break;
+        case StreamType::TEXID:
+            assert(t->target == SamplerType::SAMPLER_EXTERNAL);
+            t->gl.id = hwStream->user_thread.read[hwStream->user_thread.cur];
+            break;
+        case StreamType::ACQUIRED:
+            break;
```

Comment:
I think the behavior of the existing code here has changed and is now wrong.
Before it was checking `s->` and `hwStream->`  and now they're grouped in the same condition.